lib/charms/layer/ruby.py

```diff
@@ -1,111 +1,76 @@
+# pylint: disable=import-error
+# pylint: disable=no-name-in-module
 import os
 import sys
-from shutil import rmtree
 from multiprocessing import cpu_count
-from shell import shell
-import requests
+
+import sh
 
 from charmhelpers.core import hookenv
-from charmhelpers.core.host import chdir
-from charmhelpers.fetch import apt_install
 
 config = hookenv.config()
-WORKDIR = '/tmp/ruby'
 
 
 # HELPERS ---------------------------------------------------------------------
-def set_proxy(cmd):
-    """ Take a string to be executed by shell and add proxy if needed
+def ruby_install():
+    """ Downloads ruby-install, gpg verifies it and installs it
     """
-    http_proxy = config.get('http-proxy')
-    if http_proxy:
-        cmd = "env http_proxy='{}' {}".format(http_proxy, cmd)
-    return cmd
-
-
-def git(cmd):
-    """ simple git wrapper
-    """
-    if not os.path.isfile('/usr/bin/git'):
-        apt_install(['git'])
-    shell_cmd = set_proxy("git {}".format(cmd))
-    sh = shell(shell_cmd)
-    if sh.code > 0:
-        hookenv.status_set('blocked',
-                           'Problem with Ruby: {}'.format(sh.errors()))
+    is_snap_install = config.get("ruby-snap", None)
+    ruby_install_version = config.get("ruby-install-version", None)
+    if is_snap_install and not ruby_install_version:
+        sh.snap.install("ruby --channel {} --classic".format(config.get("ruby-snap")))
+        hookenv.status_set("maintenance", "Ruby installed from snapstore.")
+        return
```

Comment:
You need to catch 'not is_snap_install and not ruby-install-version' here, or the later steps will explode if someone set ruby_install_version to the empty string (expecting a default or latest version or similar).